dbt/task/compile.py

```diff
@@ -111,10 +135,45 @@ def __get_model_config(self, model_group, model_name):
 
         return config
 
-    def __compile(self, src_index):
-        for src_path, files in src_index.items():
+    def __find_model_by_name(self, project_models, name, package_namespace=None):
+        found = []
+        for model in project_models:
+            package, model_group, model_name = model
+            if model_name == name:
+                if package_namespace is None:
+                    found.append(model)
+                elif package_namespace is not None and package_namespace == package:
+                    found.append(model)
+
+        if len(found) == 0:
+            raise RuntimeError("Can't find a model named '{}' in package '{}' -- does it exist?".format(name, package_namespace))
+        elif len(found) == 1:
+            return found[0]
+        else:
+            raise RuntimeError("Model specification is ambiguous: model='{}' package='{}' -- {} models match criteria".format(name, package_namespace, len(found)))
+
+    def __ref(self, ctx, source_model, project_models):
+        schema = ctx['env']['schema']
+
+        # if this node doesn't have any deps, still make sure it's a part of the graph
+        self.linker.add_node(source_model)
```

Comment:
keeping `linker` stored on the CompileTask object is asking for trouble. I believe if you pass an instance of `linker` into `__ref`, it will get closed into the inner function and you'll get the desired result, without needing to store it on the CompileTask